src/Form/Control/ScopeBuilder.php

```diff
@@ -674,9 +762,34 @@ public static function conditionToQuery(Scope\Condition $condition, $inputsMap =
             'rule' => $rule,
             'operator' => $operator,
             'value' => $value,
+            'option' => self::getOption($inputType, $value, $condition),
         ];
     }
 
+    /**
+     * return extra value option associate with certain inputType or null otherwise.
+     */
+    protected static function getOption(string $type, string $value, Condition $condition): ?array
+    {
+        $option = null;
+        switch ($type) {
+            case 'lookup':
+                $model = $condition->getModel()->getField($condition->key)->reference->refModel();
+                $rec = $model->tryLoad($value);
```

Comment:
probably here `tryLoadBy($field, $value)`

Reply:
done! now use $reference->getTheirField() to properly set option value for dropdown.